nuttx-patches/stm32f7_DTCM.patch

```diff
@@ -0,0 +1,158 @@
+diff --git NuttX/nuttx/arch/arm/src/stm32f7/stm32_allocateheap.c NuttX/nuttx/arch/arm/src/stm32f7/stm32_allocateheap.c
+index 8b21ad6..ffa6d27 100644
+--- NuttX/nuttx/arch/arm/src/stm32f7/stm32_allocateheap.c
++++ NuttX/nuttx/arch/arm/src/stm32f7/stm32_allocateheap.c
+@@ -1,7 +1,7 @@
+ /****************************************************************************
+  * arch/arm/src/stm32f7/up_allocateheap.c
+  *
+- *   Copyright (C) 2015 Gregory Nutt. All rights reserved.
++ *   Copyright (C) 2015, 2017 Gregory Nutt. All rights reserved.
+  *   Author: Gregory Nutt <gnutt@nuttx.org>
+  *
+  * Redistribution and use in source and binary forms, with or without
+@@ -57,10 +57,12 @@
+ #include "up_arch.h"
+ #include "up_internal.h"
+ #include "stm32_mpuinit.h"
++#include "stm32_dtcm.h"
+ 
+ /****************************************************************************
+  * Pre-processor Definitions
+  ****************************************************************************/
++
+ /* Internal SRAM is available in all members of the STM32 family. The
+  * following definitions must be provided to specify the size and
+  * location of internal(system) SRAM:
+@@ -92,6 +94,20 @@
+ #define SRAM2_START  STM32_SRAM2_BASE
+ #define SRAM2_END    (SRAM2_START + STM32F7_SRAM2_SIZE)
+ 
++/* The STM32 F7 has DTCM memory */
++
++#undef HAVE_DTCM
++#define HAVE_DTCM 1
++#if !defined(DTCM_START) || !defined(DTCM_END)
++#  undef HAVE_DTCM
++#endif
++
++/* DTCM to be excluded from the main heap. */
++
++#ifdef CONFIG_STM32F7_DTCMEXCLUDE
++#  undef HAVE_DTCM
++#endif
++
+ /* We can't possibly have FSMC SRAM if the FSMC is not enabled */
+ 
+ #ifndef CONFIG_STM32F7_FSMC
+@@ -110,7 +126,7 @@
+ #  endif
+ #endif
+ 
+-/* There are 3 possible heap configurations:
++/* There are 4 possible heap configurations:
```

Comment:
@julianoes  good eye! I just back ported it :) - I think we will let that slide and I will catch it upstream.